File: mobile/iosApp/iosApp/Screens/Content/ContentLoadedScreen.swift
//
//  ContentLoadedScreen.swift
//  iosApp
//
//  Created by Szymon Klimek on 30.10.24.
//  Copyright © 2024 orgName. All rights reserved.
//

import app
import SwiftUI

struct ContentLoadedScreen: View {
    let state: ManageContentState.Loaded
    let actionSender: ActionSender
    
    init(state: ManageContentState.Loaded, actionSender: ActionSender) {
        self.state = state
        self.actionSender = actionSender
    }
    
    var body: some View {
        VStack {
            Text("Manage content screen loaded")
            
            Text("Has content: \(String(describing: state.hasContent))")
            
            Button("Download content") {
                actionSender.sendAction(action: ManageContentAction.DownloadUnitsTapped())
            }
        }
    }
}

File: mobile/androidApp/src/main/kotlin/com/langsapp/android/ui/content/ContentLoadedScreen.kt
package com.langsapp.android.ui.content

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.langsapp.architecture.Action
import com.langsapp.architecture.ActionSender
import com.langsapp.content.ManageContentAction
import com.langsapp.content.ManageContentState

@Composable
fun ContentLoadedScreen(
    actionSender: ActionSender<Action>,
    state: ManageContentState.Loaded,
) {
    Column {
        Text("Manage content screen loaded")
        Spacer(Modifier.height(32.dp))

        Text("Has content: ${state.hasContent}")

        Button(
            onClick = { actionSender.sendAction(ManageContentAction.DownloadUnitsTapped) },
        ) {
            Text("Download content")
        }
    }
}
